include/engine/guidance/segment_list.hpp

```diff
@@ -0,0 +1,304 @@
+#pragma once
+
+#include "osrm/coordinate.hpp"
+
+#include "engine/douglas_peucker.hpp"
+#include "engine/internal_route_result.hpp"
+#include "engine/phantom_node.hpp"
+#include "engine/segment_information.hpp"
+
+#include "extractor/turn_instructions.hpp"
+
+#include <vector>
+
+// transfers the internal edge based data structures to a more useable format
+namespace osrm
+{
+namespace engine
+{
+namespace guidance
+{
+template <typename DataFacadeT> class SegmentList
+{
+  public:
+    typedef DataFacadeT DataFacade;
+    SegmentList(const InternalRouteResult &raw_route,
+                const bool extract_alternative,
+                const unsigned zoom_level,
+                const bool allow_simplification,
+                const DataFacade *facade);
+
+    const std::vector<std::uint32_t> &GetViaIndices() const;
+    std::uint32_t GetDistance() const;
+    std::uint32_t GetDuration() const;
+
+    std::vector<SegmentInformation> const &Get() const;
```

Comment:
yeah... have to force myself for the `const T &` version each time.

I prefer `T const& name` due to me preferring `T const* name` since `const` now is directly before the part that is `const`. Especially if you have `T const* const name`. Still comes as a reflex to write it this way.